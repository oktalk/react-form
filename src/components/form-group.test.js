import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import FormGroup from './form-group'

describe(FormGroup, () => {
  const name = 'Person';
  const mockRemoveGreeting = jest.fn();
  const component = shallow(
    <FormGroup
      name={name}
      removeGreeting={mockRemoveGreeting}
    />
  );

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <FormGroup name="Person" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('contains the supplied name', () => {
    expect(component.text()).toContain(name);
  });

  it('modifies the greeting when frenchify button is clicked', () => {
    component.find('button.frenchify').simulate('click');
    expect(component.text()).toContain('Bonjour');
  });

  it('removes the greeting when remove button is clicked', () => {
    component.find('button.remove').simulate('click');
    expect(mockRemoveGreeting).toBeCalled();
  });
});
