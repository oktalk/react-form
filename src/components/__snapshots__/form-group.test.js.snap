// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class FormGroup extends _react.Component {
  constructor(props) {
    super(props);
    this.state = { greeting: 'Hello' };
    this.frenchify = this.frenchify.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
  }

  frenchify() {
    this.setState({ greeting: 'Bonjour' });
  }

  removeGreeting() {
    this.props.removeGreeting(this.props.name);
  }

  render() {
    return (
      _react2.default.createElement('div', { className: 'form-group', __source: { fileName: _jsxFileName, lineNumber: 22 }, __self: this },
        this.state.greeting, ' ', this.props.name, '!',
        _react2.default.createElement('br', { __source: { fileName: _jsxFileName, lineNumber: 24 }, __self: this }),
        _react2.default.createElement('button', { className: 'frenchify', onClick: this.frenchify, __source: { fileName: _jsxFileName, lineNumber: 25 }, __self: this }, 'Frenchify!'),
        _react2.default.createElement('button', { className: 'remove', onClick: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 26 }, __self: this }, 'Remove')));


  }} renders and matches our snapshot 1`] = `
<div
  className="form-group"
>
  Hello
   
  Person
  !
  <br />
  <button
    className="frenchify"
    onClick={[Function]}
  >
    Frenchify!
  </button>
  <button
    className="remove"
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
